src/test/java/uk/gov/register/presentation/functional/EntryResourceFunctionalTest.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.register.presentation.functional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+import io.dropwizard.jackson.Jackson;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.json.JSONException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.time.LocalDate;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class EntryResourceFunctionalTest extends FunctionalTestBase {
+    private String item1 = "{\"address\":\"6789\",\"name\":\"presley\"}";
+    private String item2 = "{\"address\":\"145678\",\"name\":\"ellis\"}";
+
+    @Before
+    public void publishTestMessages() throws Throwable {
+        dbSupport.publishMessages(ImmutableList.of(
+                String.format("{\"hash\":\"hash1\",\"entry\":%s}", item1),
+                String.format("{\"hash\":\"hash2\",\"entry\":%s}", item2)
+        ));
+    }
+
+    @Test
+    public void getEntryByEntryNumber() throws JSONException, IOException {
+        String sha256Hex = DigestUtils.sha256Hex(item1);
+
+        Response response = getRequest("/entry/1.json");
+
+        assertThat(response.getStatus(), equalTo(200));
+        assertThat(response.getHeaders().get("cache-control").toString(), equalTo("[no-transform, max-age=31536000]"));
```

Comment:
If it's a list value, can you use the `contains()` or
`containsInAnyOrder()` matchers?

On 5 April 2016 at 09:32, Om Sharma <notifications@github.com> wrote:

> In
> src/test/java/uk/gov/register/presentation/functional/EntryResourceFunctionalTest.java
> <https://github.com/openregister/presentation/pull/202#discussion_r58503787>
> :
>
> > +    @Before
> > +    public void publishTestMessages() throws Throwable {
> > +        dbSupport.publishMessages(ImmutableList.of(
> > +                String.format("{\"hash\":\"hash1\",\"entry\":%s}", item1),
> > +                String.format("{\"hash\":\"hash2\",\"entry\":%s}", item2)
> > +        ));
> > +    }
> > +
> > +    @Test
> > +    public void getEntryByEntryNumber() throws JSONException, IOException {
> > +        String sha256Hex = DigestUtils.sha256Hex(item1);
> > +
> > +        Response response = getRequest("/entry/1.json");
> > +
> > +        assertThat(response.getStatus(), equalTo(200));
> > +        assertThat(response.getHeaders().get("cache-control").toString(), equalTo("[no-transform, max-age=31536000]"));
>
> Because header value is always a list. I am ok with this but if you like I
> can write more verbose assert in here.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/openregister/presentation/pull/202/files/5f1ba9c67748a35a671e84109337d6a55bcd3619#r58503787>
>